projects/gateway/pkg/utils/proxy_transition_function.go

```
@@ -29,6 +30,6 @@ func updateDesiredStatus(original, desired *v1.Proxy) {
 	// until we make the full move.
 	equal, ok := hashutils.HashableEqual(original, desired)
 	if ok && equal {
-		desired.Status = original.GetStatus()
```

should we bubble up the error here?

---

yeah good call